connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/TestPlugins.java

```diff
@@ -98,22 +97,56 @@
+    /**
+     * Class name of a plugin which reads a version string from resource.
+     */
+    public static final String READ_VERSION_FROM_RESOURCE = "test.plugins.ReadVersionFromResource";
 
     private static final Logger log = LoggerFactory.getLogger(TestPlugins.class);
-    private static final Map<String, File> PLUGIN_JARS;
+    private static final Map<String, PluginJar> PLUGIN_JARS;
     private static final Throwable INITIALIZATION_EXCEPTION;
 
+    private static final class PluginJar {
+        String className;
+        File jarFile;
+
+        public PluginJar(String className, File jarFile) {
+            this.className = className;
+            this.jarFile = jarFile;
+        }
+
+        public String getClassName() {
+            return className;
+        }
+
+        public File getJarFile() {
+            return jarFile;
+        }
+    }
+
     static {
         Throwable err = null;
-        HashMap<String, File> pluginJars = new HashMap<>();
+        Map<String, PluginJar> pluginJars = new HashMap<>();
         try {
-            pluginJars.put(ALWAYS_THROW_EXCEPTION, createPluginJar("always-throw-exception"));
-            pluginJars.put(ALIASED_STATIC_FIELD, createPluginJar("aliased-static-field"));
-            pluginJars.put(SAMPLING_CONVERTER, createPluginJar("sampling-converter"));
-            pluginJars.put(SAMPLING_CONFIGURABLE, createPluginJar("sampling-configurable"));
-            pluginJars.put(SAMPLING_HEADER_CONVERTER, createPluginJar("sampling-header-converter"));
-            pluginJars.put(SAMPLING_CONFIG_PROVIDER, createPluginJar("sampling-config-provider"));
-            pluginJars.put(SERVICE_LOADER, createPluginJar("service-loader"));
+            pluginJars.put("always-throw-exception",
+                    new PluginJar(ALWAYS_THROW_EXCEPTION, createPluginJar("always-throw-exception")));
+            pluginJars.put("aliased-static-field",
+                    new PluginJar(ALIASED_STATIC_FIELD, createPluginJar("aliased-static-field")));
+            pluginJars.put("sampling-converter",
+                    new PluginJar(SAMPLING_CONVERTER, createPluginJar("sampling-converter")));
+            pluginJars.put("sampling-configurable",
+                    new PluginJar(SAMPLING_CONFIGURABLE, createPluginJar("sampling-configurable")));
+            pluginJars.put("sampling-header-converter",
+                    new PluginJar(SAMPLING_HEADER_CONVERTER, createPluginJar("sampling-header-converter")));
+            pluginJars.put("sampling-config-provider",
+                    new PluginJar(SAMPLING_CONFIG_PROVIDER, createPluginJar("sampling-config-provider")));
+            pluginJars.put("service-loader",
+                    new PluginJar(SERVICE_LOADER, createPluginJar("service-loader")));
+            // Create two versions of the same plugin reading version string from a resource
+            pluginJars.put("read-version-from-resource-v1",
```

Comment:
Since these string literals are now relevant elsewhere, we should make them reusable constants. Perhaps they should be enums? I realize now that perhaps the class names should have also been enums but 🤷.